features/step_definitions/collection_steps.rb

```diff
@@ -203,6 +213,14 @@
   step %{I should see /We couldn't find the collection(?:.+and)? #{collection_name}/}
 end
 
+Then("I should see {string} button exactly {int} time") do |_string, _int|
+  step %{I should see "Leave"}
+end
```

Comment:
Unfortunately, this step isn't doing what you want. It's not using either the `{string}` or `{int}` arguments it's given; it's just looking for the word "Leave" to appear anywhere on the page any number of times, no matter what it string you tell it to look for, or how many times you tell it the string should appear. 

If you want to see what I mean, try changing the feature file to say "Then I should see "Pineapple" button exactly 6 time" and then run the tests. The scenario will still pass, even though "Pineapple" doesn't appear on the collections index six times.

I'd recommend deleting lines 216-223 here and defining a new step in step_definitions/generic_steps.rb: 

```
Then "I should see {string} exactly {int} time(s)" do |string, int|
  expect(page).to have_content(string).exactly(int)
end
```

That will actually look for the specified text to appear the specified number of times. (The guts are basically an RSpec test -- if you search for `expect` in our step definitions directory, you'll find a lot of step definitions where we do that because it's just easier. I didn't know how to do it off the top of my head, either, but some determined Googling turned up [Capybara::RSpecMatchers::CountSugar](https://www.rubydoc.info/gems/capybara/Capybara/RSpecMatchers/CountSugar).)

Reply:
You're right thank you. Addressed!